From f4aec4d37447cc274b90c129ea15a008473ed02d Mon Sep 17 00:00:00 2001
From: Petr Menšík <pemensik@redhat.com>
Date: Jan 21 2021 16:30:54 +0000
Subject: Yet another change to support BIND 9.16.11 API change


Another change with another release, new parameter is added again.
Add another ifdef to keep compatibility with both versions.

---

diff --git a/src/zone.c b/src/zone.c
index d0b71b1..7ce1769 100644
--- a/src/zone.c
+++ b/src/zone.c
@@ -17,6 +17,7 @@
 #include <dns/zone.h>

 #include "util.h"
+#include "config.h"

 /**
  * Write given diff to zone journal. Journal will be created
@@ -61,7 +62,11 @@ zone_soaserial_updatetuple(dns_updatemethod_t method, dns_difftuple_t *soa_tuple
 	REQUIRE(soa_tuple->rdata.type == dns_rdatatype_soa);

 	serial = dns_soa_getserial(&soa_tuple->rdata);
+#if LIBDNS_VERSION_MAJOR >= 1611
+	serial = dns_update_soaserial(serial, method, NULL);
+#else
 	serial = dns_update_soaserial(serial, method);
+#endif
 	dns_soa_setserial(serial, &soa_tuple->rdata);
 	if (new_serial != NULL)
 		*new_serial = serial;
